File: humanresource-domain/src/main/java/com/humanresource/domain/service/JobService.java
package com.humanresource.domain.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.humanresource.domain.model.Job;
import com.humanresource.domain.repository.jpa.JobJpaRepository;

@Service
public class JobService {
	
	@Autowired
	private JobJpaRepository jobJpaRepository;
	
	public Job save(Job job) {
		return jobJpaRepository.save(job);
	}
	
	public void delete(Long id) {
		jobJpaRepository.deleteById(id);
	}
	
	public Job findOne(Long id) {
		return jobJpaRepository.findById(id).orElse(null);
		/* return jobJpaRepository.getOne(id); */
	}
	
	public List<Job> findAll() {
		return jobJpaRepository.findAll();
	}
	
	public Page<Job> findAll(int page, int count) {
		Pageable pages = PageRequest.of(page, count);
		return jobJpaRepository.findAll(pages);
	}
}

File: humanresource-web/src/app/service/job.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Job } from '../model/job';
import { LOCAL_HOST } from './constant';

@Injectable()
export class JobService {

  constructor(private http: HttpClient) { }

  create(job: Job){
    job.id = null;
    return this.http.post(`${LOCAL_HOST}/api/job`,job);
  }

  findAll(page:number,count:number){
    return this.http.get(`${LOCAL_HOST}/api/job/${page}/${count}`);
  }

  list(){
    return this.http.get(`${LOCAL_HOST}/api/job/list`);
  }

  findById(id:number){
    return this.http.get(`${LOCAL_HOST}/api/job/${id}`);
  }

  delete(id:number){
    return this.http.delete(`${LOCAL_HOST}/api/job/${id}`);
  }
}
